controllers/channel.js

```diff
@@ -413,18 +413,134 @@ exports.closeChannel = (req,res) => {
+        res.status(200).json(fill);
+    }).catch(err => {
+        global.logger.warn(err);
+        res.status(500).json({error: err});
+    });
+    ln.removeListener('error', connFailed);
+}
 
+
+//Function # 5
+//Invoke the 'listforwards' command to list the forwarded htlcs
+//Arguments - None
+/**
+* @swagger
+* /channel/listForwards:
+*   get:
+*     tags:
+*       - Channel Management
+*     name: listforwards
+*     summary: Fetch the list of the forwarded htlcs
+*     responses:
+*       200:
+*         description: channel closed successfully
+*         schema:
+*           type: object
+*           properties:
+*             in_channel:
+*               type: string
+*               description: in_channel
+*             out_channel:
+*               type: string
+*               description: out_channel
+*             in_msatoshi:
+*               type: string
+*               description: in_msatoshi
+*             in_msat:
+*               type: string
+*               description: in_msat
+*             out_msatoshi:
+*               type: string
+*               description: out_msatoshi
+*             out_msat:
+*               type: string
+*               description: out_msat
+*             fee:
+*               type: string
+*               description: fee
+*             fee_msat:
+*               type: string
+*               description: fee_msat
+*       500:
+*         description: Server error
+*/
+exports.listForwardsFilter = (req,res) => {
+    function connFailed(err) { throw err }
+    ln.on('error', connFailed);
+    var {offset, maxLen, reverse} = req.query
     //Call the listforwards command
     ln.listforwards().then(data => {
+        var forwards = data.forwards
+        if(!offset) {
+            offset = 0;
+        }
+        offset = parseInt(offset)
+        //below 2 lines will readjust the offset inside range incase they went out of it
+        offset = Math.max(offset, 0)
+        offset = Math.min(Math.max(forwards.length - 1, 0), offset)
+        if(!maxLen) {
+            maxLen = forwards.length - offset
+        }
+        maxLen = parseInt(maxLen)
+        // since length is a scalar quantity it will throw error if maxLen is negative
+        if(maxLen<0) {
+            throw Error ('maximum length cannot be negative')
+        }
+        if(!reverse) {
+            reverse = false
+        }
+        reverse = !(reverse === 'false' || reverse === false)
+        //below logic will adjust last index inside the range incase they went out
+        var lastIndex = 0
+        var firstIndex = 0
+        if(reverse === true && forwards.length !== 0) {
+            offset = (forwards.length-1) - offset
+            firstIndex = offset
+            lastIndex = Math.max(0, offset-(maxLen - 1))
+
+        } else if(reverse === false && forwards.length !== 0) {
+            firstIndex = offset
+            lastIndex = Math.min(forwards.length - 1, offset+(maxLen - 1))
```

Comment:
Shouldn't this be `firstIndex = offset + 1`? and similarly for lastIndex as well?
Ideally firstIndex and lastIndex should be pointing to the first and last item of the dataset being returned.

Reply:
offset is the start index, so if offset=5, and reverse is false, startIndex will be 5 and if reverse is true then startIndex will be (listLength-1)-offset